astrocut/cube_cut.py

```diff
@@ -424,8 +437,14 @@ def _get_cutout(self, transposed_cube, verbose=True):
             ymax = ymax_cube       
         
         # Doing the cutout
-        cutout = transposed_cube[xmin:xmax, ymin:ymax, :, :]
-    
+        if self.threads is None or self.threads > 1:
+            with ThreadPoolExecutor(max_workers=self.threads) as pool:
+                cutouts = list(pool.map(lambda x: transposed_cube[x, ymin:ymax, :, :], range(xmin, xmax)))
+            # stack the list of cutouts
+            cutout = np.stack(cutouts)
```

Comment:
I think it would help to have a few comments here on how we are dividing up the cut-out across the threads. It seems like we are slicing it by the x dimension. Does that mean we don't get much benefit from the threading if the number of columns in the cutout is small?

Reply:
I can add some comments here. This PR essentially is in place of one to astropy where I had more details of what the issue was. https://github.com/astropy/astropy/pull/14932

Reply:
I added several lines of comments that should explain things and link back to the function in astropy that we're trying to optimize

Reply:
Thanks. I took a look at that PR and I may be more confused now. I think I just don't understand the underlying data structure...